### ISSUE TITLE
Enforce Newline Between Method/Function Declarations

### ISSUE THREAD

**Issue by Vyse777** [pinterest/ktlint#2114]
## Expected Rule behavior

Ensure newline between function declarations (with closing braces '}')

## Example
When a class defines two or more methods/functions there is currently no rule enforcing a newline to exist between them. For example this is currently valid:

```kotlin
class Thing {
    fun doStuff() {
        // Do
    }
    fun doMoreStuff() {
        // Do
    }
    fun doYetAgainMoreStuff() {
        // Do
    }
}
```
Note: the lack of spaces between method definitions.

This does not collide with IntelliJ default code formatting either. In fact, when running "Reformat Code" in IntelliJ (without custom rules), it auto-adds the newlines like so:

```kotlin
class Thing {
    fun doStuff() {
        // Do
    }

    fun doMoreStuff() {
        // Do
    }

    fun doYetAgainMoreStuff() {
        // Do
    }
}
```

## Additional information
* I cannot seem to find a "best practice" anywhere in regards to Android or Kotlin formatting. In some examples containing multiple function definitions there seems to be a newline separating function definitions. Not the best example, but one nonetheless: https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods

Please let me know if I am missing anything on this post - it's my first to this repo!
I also tried to do a search for something similar before posting, let me know if I might have missed something!

**Comment by paul-dingemans** [comment]
In latest, `0.50.0` release a new experimental rule was introduced to enforce this. See https://pinterest.github.io/ktlint/0.50.0/rules/experimental/#blank-line-before-declarations.

It is indeed not explicitly documented in the style guides but it is widely adopted practice.

**Comment by Vyse777** [comment]
Ah this is awesome! I am so sorry I missed it, I looked everywhere for key words like "function definition" "closed braces" etc. etc. and couldn't find anything!

This is great, I enabled this rule and it enforced **exactly** what I am requesting. Going to close this issue!
